File: python/ampel/ampel.py
class Ampel(object):
    def __init__(self):
        self.lampeRot = False
        self.lampeGelb = False
        self.lampeGruen = False

    def setLampen(self, startwertLampeRot=True, startwertLampeGelb=False, startwertLampeGruen=False):
        self.lampeRot = startwertLampeRot
        self.lampeGelb = startwertLampeGelb
        self.lampeGruen = startwertLampeGruen

    def schalten(self):
        if (self.lampeRot, self.lampeGelb, self.lampeGruen) == (True, False, False):
            self.lampRot = True
            self.lampeGelb = True
            self.lampeGruen = False
        elif (self.lampeRot, self.lampeGelb, self.lampeGruen) == (True, True, False):
            self.lampRot = False
            self.lampeGelb = False
            self.lampeGruen = True
        elif (self.lampeRot, self.lampeGelb, self.lampeGruen) == (False, False, True):
            self.lampRot = False
            self.lampeGelb = True
            self.lampeGruen = False
        elif (self.lampeRot, self.lampeGelb, self.lampeGruen) == (False, True, False):
            self.lampRot = True
            self.lampeGelb = False
            self.lampeGruen = False
        else:
            print("Malfunctioning")

File: python/ampel/Ampel.java
public class Ampel{
        private boolean lampeRot = false;
        private boolean lampeGelb = false;
        private boolean lampeGruen = false;

        public void setLampen(boolean rot,boolean gelb,boolean gruen){
                lampeRot = rot;
                lampeGelb = gelb;
                lampeGruen = gruen;
        }

        @Override
        public String toString(){
                return "(" + lampeRot + "," + lampeGelb + "," + lampeGruen + ")";
        }

        public boolean[] gibAus(){
                boolean[] bool = new boolean[3];
                bool[0] = lampeRot;
                bool[1] = lampeGelb;
                bool[2] = lampeGruen;
                return bool;
        }

        public void schalten(){
                if(lampeRot == true && lampeGelb == false && lampeGruen == false){
                        lampeRot = true;
                        lampeGelb = true;
                        lampeGruen = false;
                } else if(lampeRot == true && lampeGelb == true && lampeGruen == false) {
                        lampeRot = false;
                        lampeGelb = false;
                        lampeGruen = true;
                } else if(lampeRot == false && lampeGelb == false && lampeGruen == true) {
                        lampeRot = false;
                        lampeGelb = true;
                        lampeGruen = false;
                } else if(lampeRot == false && lampeGelb == true && lampeGruen == false){
                        lampeRot = true;
                        lampeGelb = false;
                        lampeGruen = false;
                }else{
                        System.out.println("Malfunctioning");


                }
        }
}
